src/components/forms/ProductSelect/ProductSelect.js

```diff
@@ -0,0 +1,51 @@
+import PropTypes from 'prop-types';
+
+import AsyncSelectField from '../SelectField/AsyncSelectField';
+
+const ProductSelect = ({namePrefix, label, description, isRequired}) => {
+  const getOptions = async () => {
+    // Mocking the response
+    const response = await new Promise(resolve => {
+      setTimeout(() => {
+        resolve([
+          {
+            code: 'P1',
+            identifier: 'product1',
+            name: 'Product 1',
+          },
+          {
+            code: 'P2',
+            identifier: 'product2',
+            name: 'Product 2',
+          },
+          {
+            code: 'P3',
+            identifier: 'product3',
+            name: 'Product 3',
+          },
+        ]);
+      }, 1000);
+    });
```

Comment:
The mocking should be done in storybook, not in the component. For the component, you need to implement the real API call.

Please review the [developer documentation](https://open-formulieren.github.io/open-forms-sdk/?path=/story/developers-documentation--page#mocking-in-stories) and check an [example](https://github.com/open-formulieren/open-forms-sdk/blob/main/src/components/LanguageSelection/LanguageSelection.stories.mdx?plain=1#L61)

And an example of the API call being made: https://github.com/open-formulieren/open-forms-sdk/blob/main/src/components/LanguageSelection/LanguageSelection.js#L39